backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/SessionWithInitializedTablesFactory.java

```diff
@@ -44,17 +48,28 @@ public SessionWithInitializedTablesFactory(ClusterConfiguration clusterConfigura
     private Session createSession(Cluster cluster, String keyspace) {
         Session session = cluster.connect(keyspace);
         try {
-            new CassandraTypesCreator(module, session)
-                .initializeTypes();
-            new CassandraTableManager(module, session)
-                .ensureAllTables();
+            Stream<Boolean> operations = Stream.of(createTypes(session), createTables(session));
+            if (operations.allMatch(updated -> updated)) {
```

Comment:
This is not really explicit IMO

With what I described above you can imagine combining the tables & types status and checking it is "full", which IMO would be easier to understand.

Reply:
I agree with benoit: we should make the intent more obvious